src/assets/scripts/aircraft/Waypoint.js

```diff
@@ -14,61 +63,167 @@ import _has from 'lodash/has';
     /**
      * Initialize Waypoint with empty values, then call the parser
+     *
+     * @for Waypoint
+     * @constructor
      */
     constructor(data = {}, fms) {
         this.altitude = null;
-        this.fix      = null;
-        this.navmode  = null;
-        this.heading  = null;
-        this.turn     = null;
+        this.fix = null;
+        this.navmode = null;
+        this.heading = null;
+        this.turn = null;
         this.location = null;
         this.expedite = false;
-        this.speed    = null;
-        this.hold     = {
+        this.speed = null;
+
+        this.hold = {
             dirTurns: null,
             fixName: null,
             fixPos: null,
             inboundHd: null,
             legLength: null,
             timer: 0
         };
+
         this.fixRestrictions = {
             alt: null,
             spd: null
         };
 
+        this.route = '';
+
         this.parse(data, fms);
     }
 
     /**
      * Parse input data and apply to this waypoint
+     *
+     * @for Waypoint
+     * @method parse
+     * @param data {object}
+     * @param fms {AircraftFlightManagementSystem}
      */
     parse(data, fms) {
+        this.extractFixRestrictions(data);
+
+        this.route = _get(data, 'route', this.route);
+
         // Populate Waypoint with data
         if (data.fix) {
-            this.navmode = 'fix';
+            this.navmode = WAYPOINT_NAV_MODE.FIX;
             this.fix = data.fix;
-            this.location = window.airportController.airport_get().getFixPosition(data.fix);
+            this.location = FixCollection.getFixPositionCoordinates(data.fix);
+        }
+
+        this.setInitialNavMode();
+    }
+
+    /**
+     * @for Waypoint
+     * @method extractFixRestrictions
+     * @param fixRestrictions {object}
+     */
+    extractFixRestrictions({ fixRestrictions }) {
+        if (_isNil(fixRestrictions)) {
+            return;
+        }
+
+        this.fixRestrictions = fixRestrictions;
+    }
+
+    /**
+     * If there isn't a navmode set, set one here
+     *
+     * For aircraft that don't yet have proper guidance (eg: SID/STAR, or departing aircraft)
+     *
+     * @for Waypoint
+     * @setInitialNavMode
+     */
+    setInitialNavMode() {
+        if (this.navmode) {
+            return;
+        }
+
+        this.navmode = WAYPOINT_NAV_MODE.HEADING;
+        const airport = window.airportController.airport_get();
+        const firstRouteSegment = _head(this.route.split('.'));
+
+        if (firstRouteSegment === airport.icao && this.heading === null) {
+            // aim departure along runway heading
+            const { angle } = airport.getRunway(airport.runway);
+
+            this.heading = angle;
+        } else if (firstRouteSegment === 'KDBG' && this.heading === null) {
+            // FIXME: radial is not defined or set anywhere in this class
+            // aim arrival @ middle of airspace
+            this.heading = this.radial + Math.PI;
```

Comment:
There might be some airports that use it. Maybe MSP off the top of my head? But I'd have to check. In general, I have no qualms though; I'd love to see it gone.